apps/whale-api/src/module.api/consortium.service.ts

```diff
@@ -1,17 +1,130 @@
+    }
+  }
+
+  async getTransactionHistory (pageIndex: number, limit: number, search: string): Promise<ConsortiumTransactionResponse> {
+    const attrs = (await this.rpcClient.masternode.getGov('ATTRIBUTES')).ATTRIBUTES
+    const members: ConsortiumMember[] = []
+    const searching: boolean = search !== ''
+    const keys: string[] = Object.keys(attrs)
+    const values: object[] = Object.values(attrs)
+    const membersKeyRegex: RegExp = /^v0\/consortium\/\d+\/members$/
+    let totalTxCount: number = 0
+    let searchFound: boolean = false
+
+    keys.forEach((key: string, i: number) => {
+      if (membersKeyRegex.exec(key) !== null) {
+        const membersPerToken = values[i]
+        const memberIds: string[] = Object.keys(membersPerToken)
+        let memberDetails: Array<{ ownerAddress: string, name: string }> = Object.values(membersPerToken)
+
+        if (searching) {
+          const foundMembers = memberDetails.filter(m => m.ownerAddress === search || m.name.toLowerCase().includes(search))
+          if (foundMembers.length > 0) {
+            memberDetails = foundMembers
+            searchFound = true
+          }
+        }
+
+        for (let j = 0; j < memberDetails.length; j++) {
+          const memberId = memberIds[j]
+          if (members.find(m => m.id === memberId) === undefined) {
+            members.push({
+              id: memberId,
+              name: memberDetails[j].name,
+              address: memberDetails[j].ownerAddress
+            })
+          }
+        }
+      }
+    })
+
+    if (searching && !searchFound) {
+      const foundTx = await this.transactionMapper.get(search)
+      if (foundTx !== undefined) {
+        const transactionsOnBlock = await this.rpcClient.account.listAccountHistory('all', { maxBlockHeight: foundTx.block.height, depth: 0 })
+        const transaction = transactionsOnBlock.find(tx => tx.txid === foundTx.txid)
+
+        if (transaction === undefined) {
+          return {
+            total: 0,
+            transactions: []
+          }
+        }
+
+        return {
+          total: 1,
+          transactions: [this.formatTransactionResponse(transaction, members)]
+        }
+      }
+
+      return {
+        total: 0,
+        transactions: []
+      }
+    }
+
+    const transactions: AccountHistory[] = await this.rpcClient.account.listAccountHistory(members.map(m => m.address), {
+      txtypes: [DfTxType.MINT_TOKEN, DfTxType.BURN_TOKEN],
+      including_start: true,
+      start: pageIndex * limit,
+      limit
+    })
+
+    const promises = []
+    for (let i = 0; i < members.length; i++) {
+      promises.push(this.rpcClient.account.historyCount(members[i].address, { txtype: DfTxType.BURN_TOKEN }))
+      promises.push(this.rpcClient.account.historyCount(members[i].address, { txtype: DfTxType.MINT_TOKEN }))
+    }
+    const counts = await Promise.all(promises)
+    totalTxCount = counts.reduce((prev, curr) => {
+      return prev + curr
+    }, 0)
```

Comment:
Note: These lines can be refactored/simplified once [this](https://github.com/JellyfishSDK/jellyfish/pull/1943) PR is merged.

Reply:
Updated.